tensorflow/lite/micro/examples/micro_speech/apollo3evb/audio_provider.cc

```diff
@@ -460,11 +459,12 @@ TfLiteStatus InitAudioRecording(tflite::ErrorReporter* error_reporter) {
 #endif  // USE_TIME_STAMP
 
   // Configure, turn on PDM
+  g_pdm_dma_error_reporter = error_reporter;
   pdm_init();
   am_hal_interrupt_master_enable();
   am_hal_pdm_fifo_flush(g_pdm_handle);
   // Trigger the PDM DMA for the first time manually.
-  pdm_start_dma(g_pdm_dma_error_reporter);
+  pdm_start_dma(error_reporter);
```

Comment:
Is the problem solved with this line only?   Could you elaborate a bit more on the initialization error? Thx.

Reply:
pdm_start_dma() is the function that sets g_pdm_dma_error_reporter using the function argument.
As such the first call in InitAudioRecording() must call it with the real error_reporter (and not the uninitialized variable g_pdm_dma_error_reporter)
I wanted to keep the change small as the logic had been copied at least once for sparkfun artemis boards. Otherwise I would have moved setting g_pdm_dma_error_reporter to InitAudioRecording () and removed it from pdm_start_dma()

Reply:
Sry for coming back late. I meant you had the following code on line 462

```suggestion
  g_pdm_dma_error_reporter = error_reporter;
```

which means that even if you don't change this line (467), it will effectively behave the same. 

I think the main question is: is there a place where `g_pdm_dma_error_reporter` can't be simply replaced by error_reporter? It looks that in all places in the method, g_pdm_dma_error_reporter is either nullptr or error_reporter

Reply:
@suphoff Can you please check @wangtz's comments and keep us posted. Thanks!